TweetMate/src/main/java/com/ybsystem/tweetmate/fragments/dialog/TweetDialog.java

```diff
@@ -0,0 +1,352 @@
+                icon.setColorFilter(menu.getColor());
+                icon.setVisibility(View.VISIBLE);
+                userIcon.setVisibility(View.GONE);
+            } else {
+                // UserIcon
+                GlideUtils.load(menu.getUserIconURL(), userIcon, CIRCLE);
+                userIcon.setVisibility(View.VISIBLE);
+                icon.setVisibility(View.GONE);
+            }
+
+            // Set text
+            TextView tv = cv.findViewById(R.id.text_item);
+            tv.setText(menu.getLabel());
+
+            return cv;
+        }
+    }
+
+    private void setMenuItems(MenuArrayAdapter adapter) {
+        // Menu setting
+        Set<TweetMenu> menuSetting = PrefClickAction.getTweetMenu();
+
+        // Reply
+        if (menuSetting.contains(REPLY)) {
+            adapter.add(new Menu(
+                    R.drawable.ic_reply, COLOR_STRONG, STR_REPLY_VERB, "",
+                    () -> ClickUseCase.reply(mSource)
+            ));
+        }
+        // Retweet
+        if (menuSetting.contains(RETWEET)) {
+            if (mStatus.isRetweeted()) {
+                adapter.add(new Menu(
+                        R.drawable.ic_retweet, COLOR_RETWEET, STR_UNRETWEET, "",
+                        () -> StatusUseCase.retweet(mStatus)
+                ));
+            } else if (mSource.isPublic()) {
+                adapter.add(new Menu(
+                        R.drawable.ic_retweet, COLOR_RETWEET, STR_RETWEET, "",
+                        null
+                ));
+            }
+        }
+        // Favorite
+        if (menuSetting.contains(LIKE)) {
+            if (mStatus.isFavorited()) {
+                adapter.add(new Menu(
+                        PrefAppearance.getLikeFavDrawable(),
+                        PrefAppearance.getLikeFavColor(),
+                        PrefAppearance.getUnLikeFavText(), "",
+                        () -> StatusUseCase.favorite(mStatus)
+                ));
+            } else {
+                adapter.add(new Menu(
+                        PrefAppearance.getLikeFavDrawable(),
+                        PrefAppearance.getLikeFavColor(),
+                        PrefAppearance.getLikeFavText(), "",
+                        () -> StatusUseCase.favorite(mStatus)
+                ));
+            }
+        }
+        // Talk
+        if (menuSetting.contains(TALK)) {
+            if (mSource.isTalk()
+                    && !TweetMateApp.getActivity().getSupportActionBar().getTitle().equals(STR_TALK)
+                    && !TweetMateApp.getActivity().getSupportActionBar().getTitle().equals(STR_DETAIL)) {
+                adapter.add(new Menu(
+                        R.drawable.ic_talk, COLOR_TALK, STR_SHOW_TALK, "",
+                        () -> ClickUseCase.showTalk(mSource)
+                ));
+            }
+        }
+        // Delete
+        if (menuSetting.contains(DELETE)) {
+            if (mSource.isMyTweet() && !mStatus.isRetweet()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS1066)